src/porepy/numerics/fem/rt0.py

```diff
@@ -221,41 +240,54 @@ def massHdiv(K, c_volume, coord, sign, dim, HB):
         """
         # Allow short variable names in this function
         # pylint: disable=invalid-name
-        inv_K = np.linalg.inv(K)
-        inv_K = linalg.block_diag(*([inv_K] * (dim + 1))) / c_volume
+        I = np.eye(dim + 1)
+        # expand the inv_K tensor
+        inv_K_exp = (
+            I[:, np.newaxis, :, np.newaxis]
+            * inv_K[np.newaxis, :, np.newaxis, :]
+            / c_volume
+        )
+        inv_K_exp.shape = (I.shape[0] * inv_K.shape[0], I.shape[1] * inv_K.shape[1])
 
         coord = coord[0:dim, :]
         N = coord.flatten("F").reshape((-1, 1)) * np.ones((1, dim + 1)) - np.tile(
             coord, (dim + 1, 1)
         )
-
         C = np.diagflat(sign)
 
-        return np.dot(C.T, np.dot(N.T, np.dot(HB, np.dot(inv_K, np.dot(N, C)))))
+        return np.dot(C.T, np.dot(N.T, np.dot(HB, np.dot(inv_K_exp, np.dot(N, C)))))
 
-    @staticmethod
-    def opposite_side_node(face_nodes, nodes, faces_loc):
-        """
-        Given a face return the node on the opposite side, typical request of a Raviart-Thomas
-        approximation. This function is mainly for internal use.
+    def _compute_cell_face_to_opposide_node(self, g, data):
```

Comment:
Should it be possible to recompute by passing an argument? I.e. def _compute_cell_face_to_opposide_node(self, g, data, recompute=False)

Reply:
Can you add a short documentation at the beginning of the function? (possible heavily inspired by the documentation from the deleted function opposite_side_node)

Reply:
I agree, comments are necessary.

Reply:
okay, I've added comments now